### PR TITLE
chore(docs): partially rewrite, fix typos

### PR DESCRIPTION
**Pull request by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf**
as title.

feel free to clarify and correct me if you think I changed the intended meaning of any formulations.

wip, more to come.

### CODE REVIEW

**Comment by MarioUhrik** [comment]
Hi @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf , please don't forget to let us know when this is ready for review

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
> Hi @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf , please don't forget to let us know when this is ready for review

yeah, thanks, will `cc` you when ready; this is still a WIP.

File: docs/claudie-workflow/claudie-workflow.md [hunk 1]
```diff
@@ -29,56 +28,57 @@
...
-- Receives config with desiredState from Scheduler
+- Receives a `config` from Frontend, calculates its `msChecksum` and saves it to the database
+- Periodically checks for `config` changes and pushes the `config` to the `schedulerQueue` if `msChecksum` != `dsChecksum`
+- Periodically checks for `config` changes and pushes the `config` to the `builderQueue` if `dsChecksum` != `csChecksum`
+- Receives a `config` with the `desiredState` from Scheduler
 - Checks if `dsChecksum` == `msChecksum`
-    - `true` -> saves it to the database
-    - `false` -> it ignores this config
-- Receives config with currentState from Builder and saves it to the database
+  - `true` ⇒ ignores this `config`
```
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [inline docs/claudie-workflow/claudie-workflow.md:63]: cc @MiroslavRepka, @Despire 
could you guys, please, validate this?

the description here was using a flipped logic, which made me have a look at [the relevant code](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) and correct the documentation (I believe).

Comment (Despire) [reply docs/claudie-workflow/claudie-workflow.md:63]: [the code that you refer to](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) actually refers to the second point `Periodically checks for config changes and pushes the config to the schedulerQueue if msChecksum != dsChecksum`

Comment (Despire) [reply docs/claudie-workflow/claudie-workflow.md:63]: When I look at it the check for `dsChecksum` == `msChecksum` is missing from the [function]( https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L42-L57) I guess it was removed? @MiroslavRepka

Comment (MiroslavRepka) [reply docs/claudie-workflow/claudie-workflow.md:63]: I vaguely remember something along those lines there a while back. I guess we removed it at some point. In any way, what @Despire wrote is correct, the correct code for that section of docs is `SaveConfig(Builder|Scheduler)()`, which is called by (Builder|Scheduler) when they are done processing particular config.

Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/claudie-workflow/claudie-workflow.md:63]: > [the code that you refer to](https://github.com/Berops/claudie/blob/master/services/context-box/server/cbox.go#L99) actually refers to the second point `Periodically checks for config changes and pushes the config to the schedulerQueue if msChecksum != dsChecksum`

that's right, correct, thanks, my bad.


> When I look at it, the check for `dsChecksum` == `msChecksum` is missing from the [function](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L42-L57) I guess it was removed? @MiroslavRepka

looks like it was left up to the DB layer to handle it ([`database.UpdateDs(config)`](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L47)?
[`SaveConfigFrontend`](https://github.com/Berops/claudie/blob/master/services/context-box/server/server.go#L67-L69) still seems to perform the check, though.

bottom line is, I should probably remove the part about the "check" from the documentation, since there are none of the described kind. what do you think?

Comment (Despire) [reply docs/claudie-workflow/claudie-workflow.md:63]: Yes, I would also remove it, and just rewrite it the same as with builder, i.e. 

Receives a `config` with `currentState` from Scheduler and saves it to the database

File: docs/claudie-workflow/claudie-workflow.md [hunk 2]
```diff
@@ -29,189 +28,193 @@
...
+| `dsChecksum` | desired state checksum |
+| `csChecksum` | current state checksum |
 
 
 ## Scheduler
-Scheduler creates desired state of the infra based on the manifest taken from the config received from Context-box.
+Scheduler brings the infrastructure to a desired the state based on the manifest contained in the config that is received from Context-box.
 
-Scheduler also monitors health of the current infra and manages any operations based on it, e.g. replacement of broken node, etc. *[work in progress]*
+Scheduler also monitors the health of current infrastructure and manages any operations based on actual health state (e.g. replacement of broken nodes, etc. *[work in progress]*).
```
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [inline docs/claudie-workflow/claudie-workflow.md:81]: cc @Despire @MiroslavRepka 
there's a "work in progress" note here - do you guys know if we want to add sth here or can the note be deleted or do we want to keep it?

Comment (MarioUhrik) [reply docs/claudie-workflow/claudie-workflow.md:75]: it's still WIP as part of the autoscaler & self-healing features that are yet to come

Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/claudie-workflow/claudie-workflow.md:75]: thanks @MarioUhrik.

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
I will not be spending any more time on this, let's merge it as it is (if you see fit).

**Comment by MarioUhrik** [comment]
thanks @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf , I'd like to review this personally as soon as I have the time for it

File: docs/crud/crud.md [hunk 3]
```diff
@@ -1,45 +1,44 @@
...
+This makes users store input manifests in an [IaC](https://en.wikipedia.org/wiki/Infrastructure_as_code) manner and can easily be configured for GitOps synchronization (i.e. via FluxCD).
 
 # Update
-
-If the user wishes to update the input manifest, they can edit/reapply the secret with the updated input manifest inside of it (the secret name and the data field name will stay the same). The k8s-sidecar will notice the change in the secret data, and will update the file inside the Frontend file system. The Frontend will then apply it to Claudie and the update of the defined infrastructure will be underway.
+When you want to update the input manifest, you can edit/reapply the secret with the updated input manifest inside of it (the secret name and the data field name will stay the same). `k8s-sidecar` notices the change in the secret data and subsequently updates the file inside Frontend's file system. Frontend then applies it to Claudie and the update of the defined infrastructure is be underway.
```
Comment (MarioUhrik) [inline docs/crud/crud.md:28]: > is be underway.

probably incorrect gramatically

File: docs/input-manifest/input-manifest.md [hunk 4]
```diff
@@ -26,29 +26,30 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
 ## Providers 
 
-Contains configurations for different supported cloud providers. Atleast one provider needs to be defined out of the following supported providers 
+Contains configurations for supported cloud providers. At least one provider
+needs to be defined.
 
 - `gcp` [GCP](#gcp)
   
-  List of GCP configurations for [Google cloud](https://cloud.google.com/). This field is optional.
+  List of configurations for [Google Cloud ](https://cloud.google.com/). This field is optional.
```
Comment (MarioUhrik) [inline docs/input-manifest/input-manifest.md:34]: Might as well be `List of configuration options` like the rest.

Plus, you've added an extra whitespace after `Cloud`

File: docs/loadbalancing/loadbalancing-solution.md [hunk 5]
```diff
@@ -30,21 +30,21 @@ For creating a highly available kubernetes cluster, Loadbalancers for kubeAPI se
 
 ### Role 
 
-The Claudie uses concept or roles while configuring the loadbalancers in input manifest. Each role represents a loadbalancer configuration for a particular use. Roles are then assigned to the Loadbalancer cluster. Single loadbalancer cluster can have multiple roles assigned.
+Claudie uses the concept or roles while configuring the load balancers from the input manifest. Each role represents a loadbalancer configuration for a particular use. Roles are then assigned to the load balancer cluster. A single load balancer cluster can have multiple roles assigned.
```
Comment (MarioUhrik) [inline docs/loadbalancing/loadbalancing-solution.md:33]: `concept or roles` -> `concept of roles`

File: docs/storage/storage-solution.md [hunk 6]
```diff
@@ -2,25 +2,24 @@
 
 ## Concept
 
-Ability to run stateful workloads is a must. At the same time running stateful workloads is complex. Here the complexity is on another level considering the multi-/hybrid-cloud environment. Therefore, Claudie needs to be able to accommodate the stateful workloads, regardless of the underlying infrastructure providers.
+The ability to run stateful workloads is a must. At the same time, running stateful workloads is a complex task, even more so when considering the {multi,hybrid}-cloud environment. Therefore, Claudie needs to be able to accommodate the stateful workloads, regardless of the underlying infrastructure providers.
```
Comment (MarioUhrik) [inline docs/storage/storage-solution.md:5]: Honestly, I'd just remove this line. I feel like this sort of rubbish doesn't belong here. Thoughts ?

File: docs/storage/storage-solution.md [hunk 7]
```diff
@@ -2,25 +2,24 @@
...
-Other than default storage class, Claudie creates custom storage classes, which force persistent volumes to be created on a specific nodes based on their provider. In other words, you can use a specific provider to provision nodes for your storage needs, while using other provider for computing tasks.
+Besides the default storage class, Claudie can also create custom storage classes, which force persistent volumes to be created on a specific nodes based on the provider they have. In other words, you can use a specific provider to provision nodes for your storage needs, while using another provider for computing tasks.
 
 ## Example
 
-### [Input manifest](../input-manifest/example.yaml)
+To follow along, have a look at the reference [example input manifest file](../input-manifest/example.yaml).
 
-When Claudie will apply this input manifest, there will be a few storage classes installed. 
+When Claudie applies this input manifest, there are three storage classes installed:
```
Comment (MarioUhrik) [inline docs/storage/storage-solution.md:21]: You specify `three`, but this might change in the future, and I'd be surprised if we remembered to update this word here.
It might be better to keep it vague.

**Review by MarioUhrik — CHANGES_REQUESTED** [review]
Thanks @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf , great work polishing the docs.
I just left a few comments to think about - besides that, LGTM

File: docs/crud/crud.md (hunk 3 above)
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/crud/crud.md:28]: "is be" is an offspring of "is" and "will be"...good catch.

File: docs/storage/storage-solution.md (hunk 6 above)
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/storage/storage-solution.md:5]: this has in fact been updated quite recently. We could perhaps make it 2 sentences instead of 2..

File: docs/storage/storage-solution.md (hunk 7 above)
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/storage/storage-solution.md:21]: right, good point.

let's go with "the following"..

File: docs/loadbalancing/loadbalancing-solution.md (hunk 5 above)
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/loadbalancing/loadbalancing-solution.md:33]: THE KEYS ARE LIKE RIGHT NEXT TO EACH OTHER :D

File: docs/input-manifest/input-manifest.md (hunk 4 above)
Comment (wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf) [reply docs/input-manifest/input-manifest.md:34]: must have slipped through somehow, thanks.

**Review by MarioUhrik — APPROVED** [review]
LGTM, thanks :)

**Comment by MarioUhrik** [comment]
Any feedback before this is merged ? @MiroslavRepka @Despire

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
> I think it is good, definitely much cleaner than hooks.

hang on, wdym? @MiroslavRepka

**Comment by MiroslavRepka** [comment]
>was that perhaps intended for the other PR?

Yeah, wrong PR, mb
